### PR TITLE
Check `$this` in static closures

### PR DESCRIPTION
**Pull request by weirdan**
Fixes vimeo/psalm#9148
Fixes vimeo/psalm#2813

### CODE REVIEW

File: src/Psalm/Internal/LanguageServer/Client/TextDocument.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function xcontent(TextDocumentIdentifier $textDocument): Promise
             /**
              * @return Generator<int, Promise<object>, object, TextDocumentItem>
              */
-            static function () use ($textDocument) {
+            function () use ($textDocument) {
```
Comment (weirdan) [inline src/Psalm/Internal/LanguageServer/Client/TextDocument.php:61]: @tm1000 it looks like a number of closures used by the language server were invalid. This means the LS behaviour may change when this PR is merged, so I'd like to request your review here.

Comment (tm1000) [reply src/Psalm/Internal/LanguageServer/Client/TextDocument.php:61]: @weirdan thanks for the heads up. No idea why these were static I don't think it'll affect anything too bad. Once you merge this I'll update my WIP pr and see how it runs for a bit

**Review by tm1000 — APPROVED** [review]

File: src/Psalm/Internal/LanguageServer/Client/TextDocument.php (hunk 1 above)
Comment (weirdan) [reply src/Psalm/Internal/LanguageServer/Client/TextDocument.php:61]: It looks like all of those problematic closures were used in the code path handling `textDocument/xcontent` request. `textDocument/xcontent` is not a part of a standard protocol, it's a Sourcegraph extension, so chances are VSCode never used it.

Comment (tm1000) [reply src/Psalm/Internal/LanguageServer/Client/TextDocument.php:61]: xcontent is also removed in my pr. It existed when Felix was adding features that Microsoft hadn't yet. 

Those features are now in Microsoft's lsp spec.

**Pull request merged**